xarray/core/dataset.py

```diff
@@ -5686,5 +5686,165 @@ def map_blocks(
 
         return map_blocks(func, self, args, kwargs)
 
+    def pad(
+        self,
+        pad_width: Mapping[Hashable, Union[int, Tuple[int, int]]] = None,
+        mode: str = "constant",
+        stat_length: Union[
+            None, int, Tuple[int, int], Mapping[Hashable, Tuple[int, int]]
+        ] = None,
+        constant_values: Union[
+            None, int, Tuple[int, int], Mapping[Hashable, Tuple[int, int]]
+        ] = None,
+        end_values: Union[
+            None, int, Tuple[int, int], Mapping[Hashable, Tuple[int, int]]
+        ] = None,
```

Comment:
I'm trying to think whether we can simplify the signature of the function; it's currently really complicated and may be difficult for people to easily read a call to `pad`. (tbc, not the literal python type signature; the number of different ways to call the function).

Any thoughts? We could remove supplying `int` alone and requie a tuple? But then doesn't agree with numpy. Or require a mapping for each argument, but that seems really verbose if people want the same values for each dim (is that a common case)?

OK if the answer is "it's as complicated as it needs to be"

Reply:
Yes I think you're right...